### PR TITLE
refactor: add asset_decimals to the create_pair msg

### PR DESCRIPTION
**Pull request by kerber0x**
## Description and Motivation

This PR addresses #306, adding the asset_decimals vector to the create_pair msg so there's no need to fetch these from anywhere.

## Related Issues

#306 

---

## Checklist:

- [x] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/migaloo-core/blob/main/docs/CONTRIBUTING.md).
- [x] My pull request has a sound title and description (not something vague like `Update index.md`)
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation.
- [x] The code is formatted properly `cargo fmt --all --`.
- [x] Clippy doesn't report any issues `cargo clippy -- -D warnings`.
- [x] I have regenerated the schemas if needed `cargo schema`.

### CODE REVIEW

File: contracts/liquidity_hub/pool-manager/src/state.rs [hunk 1]
```diff
@@ -34,8 +34,6 @@ pub fn get_pair_by_identifier(
         .may_load(deps.storage, pair_identifier)?
         .ok_or(ContractError::UnExistingPair {})
 }
-// Remove after adding decimals to pair info
-pub const NATIVE_TOKEN_DECIMALS: Map<&[u8], u8> = Map::new("allow_native_token");
```
Comment (kerber0x) [inline contracts/liquidity_hub/pool-manager/src/state.rs:38]: not needed anymore

File: contracts/liquidity_hub/pool-manager/src/contract.rs [hunk 2]
```diff
@@ -174,9 +173,14 @@ fn optional_addr_validate(
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<Binary, ContractError> {
     match msg {
-        QueryMsg::NativeTokenDecimals { denom } => Ok(to_json_binary(
-            &queries::query_native_token_decimal(deps, denom)?,
-        )?),
+        QueryMsg::NativeTokenDecimals {
```
Comment (nseguias) [inline contracts/liquidity_hub/pool-manager/src/contract.rs:176]: Can we get rid of the term `Native` here?

File: contracts/liquidity_hub/pool-manager/src/contract.rs [hunk 3]
```diff
@@ -174,9 +173,14 @@ fn optional_addr_validate(
...
+            pair_identifier,
```
Comment (nseguias) [inline contracts/liquidity_hub/pool-manager/src/contract.rs:177]: Is it really needed to use both, `pair_identifier` and `denom`? Isn't `denom` unique?

File: contracts/liquidity_hub/pool-manager/src/queries.rs [hunk 4]
```diff
@@ -20,11 +21,21 @@ use crate::{math::Decimal256Helper, state::SWAP_ROUTES};
 /// Query the native token decimals
 pub fn query_native_token_decimal(
```
Comment (nseguias) [inline contracts/liquidity_hub/pool-manager/src/queries.rs:22]: maybe we can get rid of `native` term here too

File: contracts/liquidity_hub/pool-manager/src/queries.rs [hunk 5]
```diff
@@ -20,11 +21,21 @@ use crate::{math::Decimal256Helper, state::SWAP_ROUTES};
...
     deps: Deps,
+    pair_identifier: String,
     denom: String,
```
Comment (nseguias) [inline contracts/liquidity_hub/pool-manager/src/queries.rs:25]: do we need both, `pair_identifier` and `denom` here again?

File: packages/white-whale-std/src/pool_manager.rs [hunk 6]
```diff
@@ -202,7 +192,10 @@ pub enum ExecuteMsg {
 pub enum QueryMsg {
     /// Retrieves the decimals for the given native or ibc denom.
     #[returns(NativeTokenDecimalsResponse)]
```
Comment (nseguias) [inline packages/white-whale-std/src/pool_manager.rs:194]: is the `native` term needed here?

File: packages/white-whale-std/src/pool_manager.rs [hunk 7]
```diff
@@ -246,3 +239,14 @@ pub enum QueryMsg {
     #[returns(PairInfo)]
     Pair { pair_identifier: String },
 }
+
+/// The response for the `NativeTokenDecimals` query.
+#[cw_serde]
+pub struct NativeTokenDecimalsResponse {
```
Comment (nseguias) [inline packages/white-whale-std/src/pool_manager.rs:245]: why do we need to return `denom` and `decimals`? we already know them as were passed in the query.

File: packages/white-whale-std/src/pool_manager.rs (hunk 6 above)
Comment (kerber0x) [reply packages/white-whale-std/src/pool_manager.rs:194]: Good point

File: contracts/liquidity_hub/pool-manager/src/queries.rs (hunk 5 above)
Comment (kerber0x) [reply contracts/liquidity_hub/pool-manager/src/queries.rs:25]: we need them to query, but perhaps not on the response

File: contracts/liquidity_hub/pool-manager/src/contract.rs (hunk 3 above)
Comment (kerber0x) [reply contracts/liquidity_hub/pool-manager/src/contract.rs:177]: denom is not unique no, you can have duplicated pools. So you need the pair identifier to see what the decimals are for that denom on a particular pool

**Comment by kerber0x** [comment]
> Looks good overall, made some minor comments / recommendations. Tests are failing on CI

Thanks, yes the tests will keep failing, we will fix them afterwards

File: packages/white-whale-std/src/pool_manager.rs [hunk 8]
```diff
@@ -246,3 +239,14 @@ pub enum QueryMsg {
...
+/// The response for the `AssetDecimals` query.
+#[cw_serde]
+pub struct AssetDecimalsResponse {
+    /// The pair identifier to do the query for.
+    pub pair_identifier: String,
+    /// The queried denom in the given pair_identifier.
+    pub denom: String,
+    /// The decimals for the requested denom.
+    pub decimals: u8,
+}
```
Comment (kaimen-sano) [inline packages/white-whale-std/src/pool_manager.rs:252]: We should be able to delete the old `NativeTokenDecimalsResponse` as well.

File: packages/white-whale-std/src/pool_manager.rs [hunk 9]
```diff
@@ -191,18 +183,19 @@ pub enum ExecuteMsg {
...
 }
 
 #[cw_ownable_query]
 #[cw_serde]
 #[derive(QueryResponses)]
 pub enum QueryMsg {
-    /// Retrieves the decimals for the given native or ibc denom.
-    #[returns(NativeTokenDecimalsResponse)]
-    NativeTokenDecimals { denom: String },
+    /// Retrieves the decimals for the given native asset.
```
Comment (nseguias) [inline packages/white-whale-std/src/pool_manager.rs:193]: /// Retrieves the decimals for the given asset.

File: packages/white-whale-std/src/pool_manager.rs (hunk 8 above)
Comment (kerber0x) [reply packages/white-whale-std/src/pool_manager.rs:252]: Well the old `NativeTokenDecimalsResponse` is being used on the V1 contracts, so let it be for now.

**Pull request merged**